src/Symfony/Bundle/TwigBundle/DependencyInjection/TwigExtension.php

```diff
@@ -167,6 +168,26 @@ public function load(array $configs, ContainerBuilder $container): void
             }
         }
 
+        $config['cache'] ??= true;
+        if ((true === $config['cache']) && ($config['auto_reload'] ?? $config['debug'])) {
```

Comment:
When we read the configuration, the default value of `$config['debug']` is `%kernel.debug%`. It needs to be processed using `$container->getParameterBag()->resolveValue($config['debug'])`. I think `$config['auto_reload']` should also be processed with this method when it is defined.

```suggestion
        if (true === $config['cache'] && $container->getParameterBag()->resolveValue($config['auto_reload'] ?? $config['debug'])) {
```